apps/prairielearn/src/lib/cookie.ts

```diff
@@ -0,0 +1,12 @@
+import type { Request } from 'express';
+
+import { config } from './config';
+
+export function shouldSecureCookie(req: Request): boolean {
+  // In production, always set secure: true. Otherwise, only set it to true if
+  // the request is made over HTTPS.
+  //
+  // `req.protocol` should reflect Express' `trust proxy` setting, which should
+  // be used when the app is behind a reverse proxy or load balancer.
+  return !config.devMode || req.protocol === 'https';
```

Comment:
Would we want the ability to explicitly configure this? I'm thinking about an environment where an SSL load balancer is used but uses HTTP back to PL, and not in development? What would happen then?

Reply:
We do that in production, and that's what the comment above describes:

```js
  // `req.protocol` should reflect Express' `trust proxy` setting, which should
  // be used when the app is behind a reverse proxy or load balancer.
```

When running behind a load balancer, one *must* configure `trust proxy` appropriately for PL to function correctly.